core/vector_memory.py

```
@@ -10,21 +12,30 @@ def __init__(self, dim):
     def search(self, query_vector, k=5):
         """
         Return top-k metadata for nearest neighbors to query_vector.
         """
-        D, I = self.index.search(np.array([query_vector]).astype('float32'), k)
-        results = []
-        for distances, indices in zip(D, I):
-            for dist, idx in zip(distances, indices):
-                results.append((self.metadata[idx], dist))
-        return results
+        with self._lock:
+            D, I = self.index.search(np.array([query_vector]).astype('float32'), k)
```

Ruff (0.16.1)

[error] Ambiguous variable name: `I`

(E741)